File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

// food_len은 배열 food의 길이입니다.
char* solution(int food[], size_t food_len) {
    // return 값은 malloc 등 동적 할당을 사용해주세요. 할당 길이는 상황에 맞게 변경해주세요.
    int len=1;
    for (int i = 0; i < food_len; i++) {
        len += (food[i] / 2) * 2;
    }

    char* answer = (char*)malloc(sizeof(char)*(len+1));
    
    int k = 0;
    
    for (int i=1; i<food_len;i++){
        for (int j=0;j<food[i]/2;j++){
            answer[k++] = i+'0';
        }
    }
    
    answer[k++]='0';
    
    for (int i=food_len-1; i>0;i--){
        for (int j=0;j<food[i]/2;j++){
            answer[k++] = i+'0';
        }
    }
    answer[k]='\0'; 
    
    return answer;
}

File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.py
def solution(food):
    answer = ''
    
    k = 0
    for i in range(len(food)):
        for j in range(food[i]//2):
            if (i>0):
                answer+=str(i)
                
    answer+="0"
    
    for i in range(len(food)-1,0,-1):
        for j in range(food[i]//2):
            if (i>0):
                answer+=str(i)
                
    return answer